compiler/rustc_ast/src/mut_visit.rs

```diff
@@ -630,6 +630,33 @@ pub fn noop_flat_map_param<T: MutVisitor>(mut param: Param, vis: &mut T) -> Smal
     smallvec![param]
 }
 
+// No `noop_` prefix because there isn't a corresponding method in `MutVisitor`.
+pub fn visit_preexp_tt<T: MutVisitor>(tt: &mut PreexpTokenTree, vis: &mut T) {
+    match tt {
+        PreexpTokenTree::Token(token) => {
+            visit_token(token, vis);
+        }
+        PreexpTokenTree::Delimited(DelimSpan { open, close }, _delim, tts) => {
+            vis.visit_span(open);
+            vis.visit_span(close);
+            visit_preexp_tts(tts, vis);
+        }
+        PreexpTokenTree::Attributes(data) => {
+            for attr in &mut *data.attrs {
+                match &mut attr.kind {
+                    AttrKind::Normal(_, attr_tokens) => {
+                        visit_lazy_tts(attr_tokens, vis);
+                    }
+                    AttrKind::DocComment(..) => {
+                        // FIXME: Do we really need to visit doc comments?
```

Comment:
We need to visit the doc comment's span, but it's in `Attribute` and not in `AttrKind`.
Looks like `attr.span` is not currently visited.